ex04.ex

```diff
@@ -74,11 +80,15 @@ defmodule Ex04 do
   helps you do that. And, if you use that function, what does it return? That
   return value will be the thing you have to manipulate.
   """
+  # Used the ++ operator found here: https://elixir-lang.org/getting-started/basic-operators.html
+  # (note to grader: I tried to pattern match within the brackets like {even_state | val} or
+  # {[val | even_state]} but it didn't work. So I used ++)
 
-  def even_odd . . . "your code"
-
-
+  def even_odd_helper(val, {even_state, odd_state}) when Integer.is_even(val), do: {even_state ++ [val], odd_state}
```

Comment:
No deduction, but it is way more efficient to use `[ val | state ]` to build the result and then either

- reverse both output lists, or
- reverse the input list before you start.

This code will be O(n²), because it traverses the output lists for each entry in the input. Using [ h | t ] would be O(n)